### PR TITLE
workaround missing dictproxy methods Fixes #295

### LINKED ISSUE #295: AttributeError: 'dictproxy' object has no attribute 'viewitems', 'viewvalues', 'viewkeys'
```
Python 2.7.16 (default, Apr  6 2019, 01:42:57) 
[GCC 8.3.0] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> class Ham(object):
...     pass
... 
>>> import six
>>> six.viewitems(vars(Ham))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'dictproxy' object has no attribute 'viewitems'
>>> six.viewvalues(vars(Ham))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'dictproxy' object has no attribute 'viewvalues'
>>> six.viewkeys(vars(Ham))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'dictproxy' object has no attribute 'viewkeys'
```

strictly speaking this might be a cPython bug, but I think the six methods need to include a check for dictproxy and handle this case

### PR DESCRIPTION
**Pull request by graingert**

### CODE REVIEW

File: six.py [hunk 1]
```diff
@@ -607,11 +607,21 @@ def iteritems(d, **kw):
     def iterlists(d, **kw):
         return d.iterlists(**kw)
 
-    viewkeys = operator.methodcaller("viewkeys")
+    def viewkeys(d):
+        if isinstance(d, types.DictProxyType):
+            d = dict(d)
```
Comment (graingert) [inline six.py]: this should actually `return collections.abc.KeysView` etc

Comment (graingert) [reply six.py]: bluh https://github.com/benjaminp/six/issues/155

Comment (graingert) [reply six.py]: @benjaminp thoughts?

**Comment by benjaminp** [comment]
Thank for your the PR. To fix this properly, you would have to actual provide a view that was updated if the underlying dictionary was updated. Since you weren't able to view dict proxies on Python 2 anyway, I don't think this needs to be fixed, though.

**Comment by graingert** [comment]
@benjaminp ^

**Comment by benjaminp** [comment]
> collections.abc.KeysView does update and the unmoved collections.KeysView updates too

correct, but this PR does not implement that behavior for dict proxies.

**Comment by graingert** [comment]
@benjaminp, sure but I don't know how best to import collections.abc for every python version

**Comment by benjaminp** [comment]
Yes, this all leads to my feeling that the complexity of fixing this correctly is not worth it.

**Pull request closed without merging**